scripts/bootstrap/bootstrap

```diff
@@ -0,0 +1,336 @@
+[[ -z $wget_path ]] && wget_path=/home/root/.local/bin/wget
+
+# Path for the systemd unit that mounts Entware to /opt
+[[ -z $systemd_mount_path ]] && systemd_mount_path=/lib/systemd/system/opt.mount
+
+# Path where the actual Entware distribution resides (will be mounted to /opt)
+[[ -z $entware_path ]] && entware_path=/home/root/.entware
+
+# Path where the re-enable script is written
+[[ -z $reenable_path ]] && reenable_path=/home/root/entware-reenable
+
+# Toltec branch to use for this install - you almost always want 'stable'
+[[ -z $toltec_branch ]] && toltec_branch="stable"
+
+# Remove all temporary files
+cleanup() {
+    if [[ -e $wget_path ]]; then
+        rm "$wget_path"
+    fi
+}
+
+# Remove unfinished installation after unexpected error
+error-cleanup() {
+    read -ra trace < <(caller 0)
+    log ERROR "Unexpected error on line ${trace[2]}:${trace[0]} in function ${trace[1]}"
+
+    # Get out of /opt so it can be unmounted
+    cd /home/root
+
+    if [[ -d /opt ]]; then
+        umount /opt
+        rm /opt -rf
+    fi
+
+    if [[ -d $entware_path ]]; then
+        rm -rf "$entware_path"
+    fi
+
+    if [[ -f $systemd_mount_path ]]; then
+        systemctl disable --now opt.mount 2> /dev/null
+        rm "$systemd_mount_path"
+        systemctl daemon-reload
+    fi
+
+    log ERROR "This script failed to install. If you can't solve the above" \
+        "issue yourself, please report it at:" \
+        "" \
+        "   https://github.com/toltec-dev/toltec/issues" \
+        "" \
+        "(Please also include these error logs to help solving the" \
+        "problem faster. Thank you!)"
+}
+
+# Install a local wget binary which supports TLS (the original one
+# installed on the reMarkable does not) in the PATH
+wget-bootstrap() {
+    local wget_remote=https://toltec.delab.re/thirdparty/bin/wget-v1.20.3
+    local wget_checksum=8d447c6eb0a39e705f45bea900b12eef07142ea3da0809aca4dd44fe4110cdfd
+
+    if [[ ! -x $wget_path ]]; then
+        if [[ -e $wget_path ]]; then
+            log ERROR "'$wget_path' exists, but is not executable"
+            error-cleanup
+            exit 1
+        fi
+
+        # Download and compare to hash
+        mkdir -p "$(dirname "$wget_path")"
+        touch "$wget_path"
+        chmod +x "$wget_path"
+
+        wget "$wget_remote" --output-document "$wget_path" 2> /dev/null
+
+        if ! sha256sum -c <(echo "$wget_checksum  $wget_path") > /dev/null 2>&1; then
+            log ERROR "Invalid checksum for the local wget binary"
+            error-cleanup
+            exit 1
+        fi
+    fi
+
+    # Ensure the local binary is used
+    if [[ $(command -v wget) != "$wget_path" ]]; then
+        export PATH
+        PATH="$(dirname "$wget_path"):$PATH"
+    fi
+}
+
+# Mount Entware to /opt
+entware-mount() {
+    mkdir -p /opt
+    mkdir -p "$entware_path"
+
+    # Create systemd mount unit to mount over /opt on reboot
+    cat > "$systemd_mount_path" << UNIT
```

Comment:
Change this to `<<-` and you can indent the heredoc.

Reply:
But only if we indent with tabs :(